src/context.rs

```diff
@@ -0,0 +1,51 @@
+use crate::{credentials::Credentials, package::PackageStore};
+use miette::{IntoDiagnostic, Result};
+use std::path::PathBuf;
+use std::sync::Arc;
+
+/// Shared context representing `buffrs` project.
+#[derive(Debug)]
+pub struct Context {
+    /// Package store, contains installed dependencies.
+    store: PackageStore,
+    /// Credentials, used to access registries.
+    credentials: Arc<Credentials>,
```

Comment:
If we store an Arc<Context> do we need this to be an arc too?

Reply:
See the `src/resolve.rs`, there are multiple functions which take an `Arc<Credentials>`.